python/equistore-operations/equistore/operations/equal.py

```diff
@@ -81,14 +85,14 @@ def equal_block(block_1: TensorBlock, block_2: TensorBlock) -> bool:
     :param block_1: first :py:class:`TensorBlock`.
     :param block_2: second :py:class:`TensorBlock`.
     """
-    try:
-        equal_block_raise(block_1=block_1, block_2=block_2)
+    message = _equal_block_impl(block_1=block_1, block_2=block_2)
+    if message == "":
         return True
-    except NotEqualError:
+    else:
         return False
```

Comment:
Since a string is contextually convertible to bool in Python, this could also be `return not message`

Reply:
yes maybe one could do `bool(message)` I don.t have big preferences. This way it is easier to read, but takes more lines to express

Reply:
I think bool(message) is a good idea! This will remove some code as well.